### ISSUE TITLE
#line directives generated by `--cpp-lines` should preserve relative paths

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#22589]
While working on some debugging of Chapel code, I'm noticing that when compiling, say, `chpl examples/benchmarks/hpcc/ra.chpl` with `--cpp-lines`, the `#line` directives in the generated code refer to `ra.chpl` rather than `examples/benchmarks/hpcc/ra.chpl` (and similarly for internal/standard modules).  This has the effect, within the debugger, of trying to look at the sources using `$CWD/ra.chpl` (say) which is not accurate.  It seems like it would be better if we were to generate whatever relative or absolute path we have to the file in the `#line` directive instead (e.g., `#line 23 examples/benchmarks/hpcc/ra.chpl`).

I suspect that probably reflects that I've/we've only really leaned on these filenames when debugging simple programs that happen to be in the current, local directory.

**Comment by jabraham17** [comment]
I found this a few weeks ago and meant to file it as a bug, I guess I forgot. It was frustrating enough that I tracked down the code that did it. This line [here](https://github.com/chapel-lang/chapel/blame/4bbe5e73a2bdd69807e643282ddb1833df238a1c/compiler/backend/beautify.cpp#L215) is supposed to avoid lines being too long, but also removes absolute path information. Removing the `strrchr` and the if statement after should fix it
